### PR TITLE
DOC: Corrected Out-of-date list of sort algorithms

### LINKED ISSUE #38287: DOC: Out-of-date list of sort algorithms
#### Location of the documentation

e.g. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.sort_values.html, https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_index.html

#### Documentation problem

List of values for `kind` should be updated to reflect changes in NumPy: https://numpy.org/doc/stable/reference/generated/numpy.sort.html

Specifically, since NumPy 1.15.0, the list has been `{‘quicksort’, ‘mergesort’, ‘heapsort’, ‘stable’}`

#### Suggested fix for documentation

Full list of options is provided. Also, the statement that "‘mergesort’ is the only stable algorithm" is now incorrect.

I can do this, when I get time. But I'm more than happy for someone else to do it if they want.

### PR DESCRIPTION
**Pull request by FullBeardDev**
Reference Issues/PRs
Fixes: #38287

Implemented changes:
Changed the docstring of all the methods using np.sort as it now includes one additional stable algorithm

### CODE REVIEW

**Comment by jreback** [comment]
can you also update the tests in test_nargsort to add 'stable' in pandas/tests/test_sorting.py

**Comment by DonPablo99** [comment]
> can you also update the tests in pandas/tests/series/methods/test_sort_index.py to use stable (where we use mergesort), just parameterize a test is prob good enough.

Done :)

**Comment by mroeschke** [comment]
LGTM, mind fixing up the merge conflict @DonPablo99?

File: pandas/tests/series/methods/test_sort_index.py [hunk 1]
```diff
@@ -104,18 +104,13 @@ def test_sort_index_multiindex(self, level):
         res = s.sort_index(level=level, sort_remaining=False)
         tm.assert_series_equal(s, res)
 
-    def test_sort_index_kind(self):
+    @pytest.mark.parametrize("kind", ["quicksort", "mergesort", "heapsort", "stable"])
```
Comment (jreback) [inline pandas/tests/series/methods/test_sort_index.py:107]: can you make this a fixture (put in the top of this test file is fine) as well.

**Comment by jreback** [comment]
can you merge master and update

File: pandas/tests/series/methods/test_sort_index.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/series/methods/test_sort_index.py:107]: can you do this

**Review by jreback — CHANGES_REQUESTED** [review]
comment on testing. ping on green.

File: pandas/tests/series/methods/test_sort_index.py (hunk 1 above)
Comment (DonPablo99) [reply pandas/tests/series/methods/test_sort_index.py:107]: Should I just add a fixture or add it and replace the parametrize statements ?

Comment (jreback) [reply pandas/tests/series/methods/test_sort_index.py:107]: add a fixture in this file and use it

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @DonPablo99

**Pull request merged**